src/main/java/org/mockito/internal/invocation/InvocationMatcher.java

```diff
@@ -141,13 +141,21 @@ private void captureVarargsPart(Invocation invocation) {
         for (ArgumentMatcher m : uniqueMatcherSet(indexOfVararg)) {
             if (m instanceof CapturesArguments) {
                 Object rawArgument = invocation.getRawArguments()[indexOfVararg];
-                for (int i = 0; i < Array.getLength(rawArgument); i++) {
-                    ((CapturesArguments) m).captureFrom(Array.get(rawArgument, i));
+                for (int i = 0; i < getLength(rawArgument); i++) {
```

Comment:
true,
however code with if-else will be longer - but you know - there are no perfect solutions.